libs/wingsdk/src/core/tokens.ts

```diff
@@ -1,33 +1,45 @@
 import { IInflightHost } from "../std";
 
-/**
- * Represents values that can only be resolved after the app is synthesized.
- * Tokens values are captured as environment variable, and resolved through the compilation target token mechanism.
- */
-export abstract class Tokens {
+export interface ITokenResolver {
   /**
    * Returns true is the given value is a token.
    */
-  public abstract isToken(value: any): boolean;
+  isToken(value: any): boolean;
 
   /**
    * "Lifts" a value into an inflight context.
    */
-  public abstract lift(value: any): string;
+  lift(value: any): string;
 
   /**
    * Lifts the given token to the host.
    */
-  public abstract onLiftValue(host: IInflightHost, value: any): void;
+  onLiftValue(host: IInflightHost, value: any): void;
+}
+
+/**
+ * Represents values that can only be resolved after the app is synthesized.
+ * Tokens values are captured as environment variable, and resolved through the compilation target token mechanism.
+ */
+export abstract class Tokens {
```

Comment:
Why is this abstract? Seems like all methods are static.

You can use a private constructor to protect against instantiation if that was the intent.

Reply:
Just some messy get-it-working code based on the existing system, I have it as a todo in the description